src/pint/utils.py

```diff
@@ -1595,3 +1596,38 @@ def print_color_examples():
                     end="",
                 )
             print("")
+
+
+class GroupIterator:
```

Comment:
This might be easier to write as a generator:
```python
def group_iterator(items):
    for item in np.unique(items):
        yield self.items==item
```
Shorter, and it doesn't generate or keep the indexing arrays except when needed.

Reply:
good idea.  I've done that, although for now the interface is the same.